src/generateId.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Generate a pseudo-unique ID on each function call
+ *
+ * @returns {number}
+ */
+
+const generateId = (() => {
+  let counter = 0;
+
+  return () => counter++;
```

Comment:
Digging the incrementing id! What about turning the return value into a string, with a prefix, to reduce a conflict? Something like: () => \`scrollto-{counter++}\`.

Reply:
@ganderzz 👍 I made the update.

I think seeing a prefix of `scrollto-1` is better for debugging purposes than just `1`.

Where are you concerned about a possible name conflict? I'm just curious if I'm missing anything.

Reply:
Mostly I can see a conflict if multiple scroll areas are rendered.

```jsx
render() {
    return (
        <ScrollTo>
        {
        (scroll, scrollById) => (
             <>
                <ScrollArea id={0}>
                </ScrollArea>

                <ScrollArea>
                </ScrollArea>
             </>
        )
        }
        </ScrollTo>
    );
}
```

We can set the first id to 0, but when the second comes around, Id will equal null. GenerateID will do its thing and the first value will be 0--overwriting the first component. Probably an off-case, but I can see `id` being used like `key` prop in a loop. A lot of people will just throw in the iterating, `i`, value.